src/app/delivery-history/delivery-history.component.ts

```diff
@@ -65,15 +66,47 @@ export class DeliveryHistoryComponent implements OnInit {
 
     this.helperInfo = context.for(helperHistoryInfo, this.helperStorage).gridSettings({
       showFilter: true,
+      allowSelection: true,
 
       numOfColumnsInGrid: (this.settings.isSytemForMlt() ? 10 : 7),
       gridButtons: [{
-        name: this.settings.lang.exportToExcel,
-        visible: () => this.context.isAllowed(Roles.admin),
-        click: async () => {
-          await saveToExcel(this.settings, this.context.for(helperHistoryInfo, this.helperStorage), this.helperInfo, this.settings.lang.volunteers, this.busy, (d: helperHistoryInfo, c) => c == d.courier);
+          name: this.settings.lang.exportToExcel,
+          visible: () => this.context.isAllowed(Roles.admin),
+          click: async () => {
+            await saveToExcel(this.settings, this.context.for(helperHistoryInfo, this.helperStorage), this.helperInfo, this.settings.lang.volunteers, this.busy, (d: helperHistoryInfo, c) => c == d.courier);
+          }
+        },
+        {
+          name: 'הענק מתנה',
+          visible: () => this.settings.isSytemForMlt() && this.context.isAllowed(Roles.admin),
+          click: async () => {
+            let rows = this.helperInfo.selectedRows;
+            
+            if(rows.length == 0) {
+              this.dialog.Error('לא נבחרו מתנדבים');
+              return;
+            }
+
+            if (await this.context.openDialog(YesNoQuestionComponent, q => q.args = {
+              question: 'האם להעניק מתנה ל ' + rows.length + ' מתנדבים?'
+            }, q => q.yes)) {
+              if(await context.for(HelperGifts).count(g=>g.assignedToHelper.isEqualTo('')) >= rows.length) {
+                try {
+                  for await (const h of rows) {
+                    await HelperGifts.assignGift(h.courier.value);
+                  }
+                  this.refresh();
+                }
+                catch (err) {
```

Comment:
Why do you need this try catch frase - it's implicit, even if you don't catch it a dialog with an error will appear and display the message.
I prefer to handle such concerns in one place, and not have to write try catch throughout my code.

If you didn't like the way the error is displayed, you are welcome to fix it in:
https://github.com/noam-honig/hug-moms-even-yehuda/blob/aacfdc2c5e4f6c3dab12f139631521968e071cb5/src/app/select-popup/dialog.ts#L208